### PR TITLE
Toggle shared payment partial

### PR DESCRIPTION
**Pull request by sauloperez**
#### What? Why?

Related to #6908

Adds the necessary integration tests to properly check we feature-toggle the `shared/_payment.html.haml`, which is used in some mailers.

I chose to have a separate set of methods that implement the new calculation because the toggle is context-specific and as such we can't toggle at the model level. This is up to the caller. So, toggling will switch to one or the other set of methods ensuring all shown balances are consistent across the app.

:warning: this depends on #6879, so please, read from `Unit-test Spree::Order#display_outstanding_balance` onwards.

#### Release notes

Changelog Category: User facing changes | Technical changes

### CODE REVIEW

File: engines/order_management/app/services/order_management/order/updater.rb [hunk 1]
```diff
@@ -157,7 +157,7 @@ def infer_payment_state
       def infer_payment_state_from_balance
         # This part added so that we don't need to override
         # order.outstanding_balance
-        balance = order.outstanding_balance
+        balance = OrderBalance.new(order).to_f
         balance = -1 * order.payment_total if canceled_and_paid_for?
```
Comment (Matt-Yorkley) [inline engines/order_management/app/services/order_management/order/updater.rb:161]: Will this need to be updated now that the balance will be different for cancelled orders?

Comment (sauloperez) [reply engines/order_management/app/services/order_management/order/updater.rb:161]: right! nicely spot @Matt-Yorkley I was blindly replacing calls to `#outstanding_balance` and missed this

File: app/models/order_balance.rb [hunk 2]
```diff
@@ -21,6 +21,10 @@ def to_f
     end
   end
 
+  def to_html
+    "#{label}:<strong>#{amount}</strong>".html_safe
```
Comment (Matt-Yorkley) [inline app/models/order_balance.rb:25]: I feel like wrapping the output in `<strong>` tags to make it **bold** shouldn't be a decision that's made in the model, it should stay in the view layer, right?

Comment (sauloperez) [reply app/models/order_balance.rb:25]: Yes, although I thought it wasn't worth extracting something else. In the end, it's just as convenient as with `Money`'s #to_html

**Pull request closed without merging**